src/Parser/RetributionPaladin/Modules/Items/Tier20_4set.js

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatNumber } from 'common/format';
+
+import Module from 'Parser/Core/Module';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class Tier20_4set extends Module {
+	static dependencies = {
+		combatants: Combatants,
+	};
+
+	holyPowerGained = 0;
+
+	on_byPlayer_cast(event) {
+		if(this.combatants.selected.hasBuff(SPELLS.RET_PALADIN_T20_4SET_BONUS_BUFF.id) && (event.ability.guid === SPELLS.BLADE_OF_JUSTICE.id || event.ability.guid === SPELLS.DIVINE_HAMMER_TALENT.id)){
```

Comment:
This is a huge `if` statement, can you abstract your logic for determining a valid cast into a method? Something like `benefitsFrom4Pc`.